### ISSUE TITLE
Inconsistencies in triAF model calculations

### ISSUE THREAD

**Issue by mducle** [SpinW/spinw#148]
### Issue

Running successive spin wave calculations on the same `triAF` model sometimes generates spectra which disagree in a few bins.

### To reproduce

Run the following script:

```matlab
q_start = [0, 0, 0];
q_end = [1, 1, 0];
pts = 501;
evect = linspace(0.5, 5, 500);
model = 'triAF';

s = sw_model(model, 1);
spec = sw_egrid(sw_neutron(spinwave(s, {q_start, q_end, pts})), 'component', 'Sperp', 'Evect', evect);

s2 = sw_model(model, 1);
spec2 = sw_egrid(sw_neutron(spinwave(s2, {q_start, q_end, pts})), 'component', 'Sperp', 'Evect', evect);

fprintf('\nabsolute delta %f\n', max(abs(spec.swConv(:) - spec2.swConv(:))))
dd = spec.swConv - spec2.swConv; idx = find(abs(dd)>0.01);
[dd(idx) spec.swConv(idx) spec2.swConv(idx)]

figure; pcolor(dd*10+spec.swConv); shading flat; colorbar
```

Sometimes it reports no discrepancies, sometimes intensities are transposed between neighbouring bins (can be seen in the figure).

If the `chain` or `squareAF` model are used instead of `triAF` then `swConv` always agree between the two runs. There seems to be something to do with the incommensurate nature of the `triAF` model.

**Comment by RichardWaiteSTFC** [comment]
I'll check whether this has been fixed by #130

**Comment by RichardWaiteSTFC** [comment]
Yeh this is still a problem...FYI the discrepancy occurs in bins indicated with a red cross in this image
![image](https://github.com/SpinW/spinw/assets/55979119/32c99e5f-e3a4-4889-a7ad-c5d221e786e3)

**Comment by mducle** [comment]
Yeah... to be honest its small enough it usually not noticeable - except it might cause flaky tests if we didn't have such a large tolerance on the tests.

Also, I'm not entirely sure how to fix it...

So, we might have to leave this until after release 4.0
